ReactNativeClient/setUpQuickActions.ts

```diff
@@ -28,25 +32,25 @@ export default (dispatch: Function, folderId: string) => {
 		// note. If you navigate around enough (which I think changes the redux
 		// state sufficiently or something), then it'll work again.
 		dispatch({ type: 'NAV_BACK' });
+		dispatch({ type: 'SIDE_MENU_CLOSE' });
 
-		if (data.type === 'New note') {
-			dispatch({
-				type: 'NAV_GO',
-				noteId: null,
-				folderId,
-				routeName: 'Note',
-				itemType: 'note',
-			});
-		}
+		const isTodo = data.type === 'New to-do' ? 1 : 0;
 
-		if (data.type === 'New to-do') {
+		Note.save({
+			parent_id: folderId,
+			is_todo: isTodo,
+		}, { provisional: true }).then((newNote: any) => {
 			dispatch({
 				type: 'NAV_GO',
-				noteId: null,
+				noteId: newNote.id,
 				folderId,
 				routeName: 'Note',
-				itemType: 'todo',
 			});
-		}
-	});
+		});
+	};
+
+	DeviceEventEmitter.addListener('quickActionShortcut', handleQuickAction);
+
+	QuickActions.popInitialAction().then(handleQuickAction).catch((reason: any) => reg.logger().error(reason));
```

Comment:
Do you know what's the format of `reason`? Because sometimes React Native pass around objects that contain huge structures that could potentially crash the app. I think it would be safer to create a custom message like `"QuickActions.popInitialAction: Error " + reason.message` but that depends on the structure of "reason".

Reply:
It's not documented and I could not reproduce it but from the library code I can see that on Android it's just an error message:
https://github.com/jordanbyron/react-native-quick-actions/blob/21c852880222ab9cedbaab3e83e887ba8d86740f/android/src/main/java/com/reactNativeQuickActions/AppShortcutsModule.java#L78-L78

On iOS it looks like it will never return an error:
https://github.com/jordanbyron/react-native-quick-actions/blob/13c61befb94c48344d25176f4e11b583f6c6ca35/index.ios.js#L5-L18

Reply:
Thanks for checking, let's leave it as it is then.